discovery-provider/plugins/notifications/src/__tests__/notification.test.ts

```diff
@@ -1,41 +1,239 @@
-import { expect, test } from '@jest/globals';
+import { expect, jest, test } from '@jest/globals';
 import { Processor } from '../main';
 import * as sns from '../sns'
-import { clearAllTables, createUser, insertFollow, insertMobileDevice, insertMobileSetting } from '../utils/populateDB';
+import { getRedisConnection } from './../utils/redisConnection'
+import { config } from './../config'
+import { randId, clearAllTables, createChat, readChat, insertMessage, insertReaction, setupTwoUsersWithDevices } from '../utils/populateDB';
 
-
-test("Start processor", async () => {
+describe('Notification processor', () => {
+  let processor: Processor
   const sendPushNotificationSpy = jest.spyOn(sns, 'sendPushNotification')
     .mockImplementation(() => Promise.resolve())
 
-  console.log('start processor')
-  const processor = new Processor()
-  await processor.init()
-  console.log('inserting')
-  await clearAllTables(processor.discoveryDB)
-  await clearAllTables(processor.identityDB)
-  await createUser(processor.discoveryDB, { user_id: 1 })
-  await createUser(processor.discoveryDB, { user_id: 2 })
-  await insertFollow(processor.discoveryDB, 1, 2)
-  await insertMobileSetting(processor.identityDB, 2)
-  await insertMobileDevice(processor.identityDB, 2)
-  await new Promise(resolve => setTimeout(resolve, 10))
-
-  const pending = processor.listener.takePending()
-  expect(pending?.appNotifications).toHaveLength(1)
-  // Assert single pending
-  await processor.appNotifications.process(pending.appNotifications)
-
-  expect(sendPushNotificationSpy).toHaveBeenCalledWith({
-    type: 'ios',
-    targetARN: 'arn:2',
-    badgeCount: 0
-  }, {
-    title: 'Follow',
-    body: 'user_1 followed you',
-    data: {}
+  beforeEach(async () => {
+    const redis = await getRedisConnection()
+    redis.del(config.lastIndexedMessageRedisKey)
+    redis.del(config.lastIndexedReactionRedisKey)
+    console.log('init processor')
+    processor = new Processor()
+    await processor.init()
+
+    await clearAllTables(processor.discoveryDB)
+    await clearAllTables(processor.identityDB)
+  })
+
+  afterEach(async () => {
+    jest.clearAllMocks()
+    await processor.listener.close()
+    processor.stop()
+  })
+  
+  // test("Sends app notifications", async () => {
```

Comment:
I'll fix this - am adding the ability to create a new db from a template to run all test in parallel